### PR TITLE
Generic Gateway Events + Service Reformatting

### PR DESCRIPTION
**Pull request by hyperschwartz**
# Description
This PR includes the concept of dynamic gateway events that can be emitted from any smart contract.  They are verified using the event structure cross-referenced with the referenced scope and target addresses to ensure that the events are sent from the correct source.  Additionally, this PR introduces the concept of access revocations using the same dynamic event structure.

## Changes
- Cleanup on various build scripts
- Introduce ktlint to the build for kotlin sub-projects
- Introduce pre-commit hook for ktlint checks
- Introduce adarshr's test logger for prettier test output in builds
- Remove staging repository sonatype resolution now that we only consume primary repository sources
- Remove contract address link in server properties - it wasn't used, and now with dynamic events, it shouldn't be used
- Remove unused legacy event stream configurations
- Create a `GatewayEventAdapter` to use dynamically with the asset classification events + gateway events.  This will possibly be merged with the `GatewayEvent` class that utilizes it once asset classification events are no longer processed by the application
- Add a `revokeAccessPermission` function to the `ScopePermissionsRepository` that removes all records that grant access to a grantee for a scope address
- Refactor `StreamEventHandlerService` to process both gateway events and asset classification events based on the event contents.  This flow favors gateway events as the target source, and then falls back to asset classification events if the incoming event does not fit the gateway event structure

### CODE REVIEW

File: server/src/main/kotlin/io/provenance/objectstore/gateway/eventstream/AcContractKey.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package io.provenance.objectstore.gateway.eventstream
 
-enum class ContractKey(val eventName: String) {
+// TODO: Remove this class when asset classification events are no longer consumed
+enum class AcContractKey(val eventName: String) {
```
Comment (hyperschwartz) [inline server/src/main/kotlin/io/provenance/objectstore/gateway/eventstream/AcContractKey.kt:4]: Renamed to better distinguish this from gateway stuff

File: server/src/main/kotlin/io/provenance/objectstore/gateway/service/StreamEventHandlerService.kt [hunk 2]
```diff
@@ -15,29 +23,127 @@ class StreamEventHandlerService(
...
+        // This ensures that properly-stored scopes will be able to be successfully decrypted in future requests by
+        // the granted address.
+        val granterAddress = findRegisteredScopeOwnerAddress(scopeResponse) ?: run {
+            logger.info("Skipping $gatewayEvent not owned by any registered granters")
+            return
+        }
+        // Ensure that the scope's value owner signed for the transaction that emitted this wasm event.  This ensures
+        // that this access grant has been safely requested and should be made
+        getSignerAddressesForTx(gatewayEvent.txHash).also { signerAddresses ->
+            if (scopeResponse.scope.scope.valueOwnerAddress !in signerAddresses) {
```
Comment (piercetrey-figure) [inline server/src/main/kotlin/io/provenance/objectstore/gateway/service/StreamEventHandlerService.kt:66]: We should probably have a TODO in here about adding a fallback check for an authz contract execute grant from the valueOwnerAddress to any of the signerAddresses... I think we will need that, right? In the case where someone delegates the classification of their scopes for them, for instance?

Comment (hyperschwartz) [reply server/src/main/kotlin/io/provenance/objectstore/gateway/service/StreamEventHandlerService.kt:66]: I totally forgot about that and you're absolutely right.  I'll pop that TODO on there

Comment (hyperschwartz) [reply server/src/main/kotlin/io/provenance/objectstore/gateway/service/StreamEventHandlerService.kt:68]: Done did the thing 👍

**Review by piercetrey-figure — APPROVED** [review]

**Pull request merged**